cli/lsp/diagnostics.rs

```diff
@@ -127,12 +146,13 @@ impl DiagnosticsServer {
   pub(crate) fn start(
     &mut self,
     language_server: Arc<Mutex<language_server::Inner>>,
-    client: Client,
-    ts_server: Arc<tsc::TsServer>,
```

Comment:
I am hoping to get rid of how `language_server` is passed around to functions like this in the future and this PR is a stepping stone for that.